### PR TITLE
add acceptance

### PR DESCRIPTION
**Pull request by johngruen**

### CODE REVIEW

File: docs/acceptance.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+- [ ] It should have a feedback button that links to an external survey with the query params `ver={add-on-version}`, `ref={sidebar}` and `rel={firefox-release}`
+- [ ] It should be titled with the extension name
+- [ ] It should include the extension icon in the header
+
+### Sidebar WebView
+- [ ] It should include a back button to return to the sidebar homepage
+- [ ] It should inlude a refresh button to refresh the page
+- [ ] It should render websites using a common mobile header
+- [ ] It should let users switch to a normal desktop header
+- [ ] It should remember if users make the choice to switch headers
```
Comment (wresuolc) [inline docs/acceptance.md]: per site or global?

File: docs/acceptance.md [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+### Context Clicks
+- [ ] It should register a _link_ context menu item for opening sites in the sidebar
+- [ ] It should register a _page_ and _tab_ context menu item for opening sites in the sidebar
+- [ ] Both context menu buttons should include the extension icon
+
+### Sidebar Homepage
+- [ ] It should have a homepage
+- [ ] It should have disimissable onboarding content (text + icon for MVP)
+- [ ] It should show a list of currently open tabs
+- [ ] It should show a list of recently sidebarred tabs
```
Comment (wresuolc) [inline docs/acceptance.md]: I think this needs a small `clear` link to empty the list.  Also, call out ignoring private mode urls too?

File: docs/acceptance.md [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+- [ ] It should include a back button to return to the sidebar homepage
+- [ ] It should open clicked links in a new tab in the active window
+- [ ] It should inlude a refresh button to refresh the page
+- [ ] It should render websites using a common mobile header
+- [ ] It should let users switch to a normal desktop header
+- [ ] It should remember per site if users make the choice to switch headers 
+
+### Global
+- [ ] It should extend the max-width of the sidebar to up to 50% of the total browser window
+
```
Comment (wresuolc) [inline docs/acceptance.md]: L10n here?  Or in stretch?

File: docs/acceptance.md [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+### Stretch
+- [ ] The sidebar tab should be openable by a browserAction. Currently blocked by[this bug](https://bugzilla.mozilla.org/show_bug.cgi?id=1427431).
```
Comment (wresuolc) [inline docs/acceptance.md]: This bug is verified:fixed in 59.  is this the right bug?

File: docs/acceptance.md [hunk 5]
```diff
@@ -0,0 +1,42 @@
+<img width="840px" height="auto" src="./acceptance-wires.png" alt="Splitting basics">
+
+### In General
+- [ ] It should generally match Photon styles.
+- [ ] It should have a way for users to provide feedback without leaving the experiment.
```
Comment (ianb) [inline docs/acceptance.md]: #6 - that bug currently needs a destination link

File: docs/acceptance.md [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+- [ ] It should match final visual specs.
+
+### Context Clicks
+- [ ] It should register a _link_ context menu item for opening sites in the sidebar
```
Comment (ianb) [inline docs/acceptance.md]: #3 (implemented)

File: docs/acceptance.md [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+- [ ] It should register a _page_ and _tab_ context menu item for opening sites in the sidebar
```
Comment (ianb) [inline docs/acceptance.md]: #4 (implemented). Note there's also #16, which is to add a context menu to bookmarks (should be simple)

File: docs/acceptance.md [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+- [ ] Both context menu buttons should include the extension icon
```
Comment (ianb) [inline docs/acceptance.md]: Blocked on #14

File: docs/acceptance.md [hunk 9]
```diff
@@ -0,0 +1,42 @@
...
+
+### Sidebar Homepage
+- [ ] It should have a homepage
```
Comment (ianb) [inline docs/acceptance.md]: Mostly #9? Needs some followup bugs/design

File: docs/acceptance.md [hunk 10]
```diff
@@ -0,0 +1,42 @@
...
+- [ ] It should have disimissable onboarding content (text + icon for MVP)
+- [ ] It should show a list of currently open tabs
```
Comment (ianb) [inline docs/acceptance.md]: #2 (fixed in #18)

File: docs/acceptance.md [hunk 11]
```diff
@@ -0,0 +1,42 @@
...
+- [ ] It should show a list of recently sidebarred tabs
+- [ ] It should have a feedback button that links to an external survey with the query params `ver={add-on-version}`, `ref={sidebar}` and `rel={firefox-release}`
```
Comment (ianb) [inline docs/acceptance.md]: `ref=sidebar` is a static string, right?

File: docs/acceptance.md [hunk 12]
```diff
@@ -0,0 +1,42 @@
...
+- [ ] It should be titled with the extension name
+- [ ] It should include the extension icon in the header
+
+### Sidebar WebView
+- [ ] It should include a back button to return to the sidebar homepage
```
Comment (ianb) [inline docs/acceptance.md]: #9 (currently labeled "home")

File: docs/acceptance.md [hunk 13]
```diff
@@ -0,0 +1,42 @@
...
+- [ ] It should open clicked links in a new tab in the active window
+- [ ] It should inlude a refresh button to refresh the page
```
Comment (ianb) [inline docs/acceptance.md]: #11 (probably feasible, but I have some concerns)

File: docs/acceptance.md [hunk 14]
```diff
@@ -0,0 +1,42 @@
...
+- [ ] It should render websites using a common mobile header
```
Comment (ianb) [inline docs/acceptance.md]: #7 (already closed)

File: docs/acceptance.md [hunk 15]
```diff
@@ -0,0 +1,42 @@
...
+- [ ] It should let users switch to a normal desktop header
```
Comment (ianb) [inline docs/acceptance.md]: #8 (fixed in #18)

File: docs/acceptance.md [hunk 16]
```diff
@@ -0,0 +1,42 @@
...
+- [ ] It should remember per site if users make the choice to switch headers
```
Comment (ianb) [inline docs/acceptance.md]: Also fixed in #18

File: docs/acceptance.md [hunk 17]
```diff
@@ -0,0 +1,42 @@
...
+### Sidebar WebView
+- [ ] It should include a back button to return to the sidebar homepage
+- [ ] It should open clicked links in a new tab in the active window
+- [ ] It should inlude a refresh button to refresh the page
+- [ ] It should render websites using a common mobile header
+- [ ] It should let users switch to a normal desktop header
+- [ ] It should remember per site if users make the choice to switch headers 
+
+### Global
+- [ ] It should extend the max-width of the sidebar to up to 50% of the total browser window
```
Comment (ianb) [inline docs/acceptance.md]: #10

File: docs/acceptance.md (hunk 4 above)
Comment (ianb) [inline docs/acceptance.md]: #23

File: docs/acceptance.md [hunk 18]
```diff
@@ -0,0 +1,42 @@
...
+
+### Stretch
+- [ ] The sidebar tab should be openable by a browserAction. Currently blocked by[this bug](https://bugzilla.mozilla.org/show_bug.cgi?id=1427431).
+- [ ] The browser action should accept a fill value of `context-fill`
```
Comment (ianb) [inline docs/acceptance.md]: I think you mean the browser action *icon*

File: docs/acceptance.md [hunk 19]
```diff
@@ -0,0 +1,42 @@
...
+- [ ] It should match final visual specs.
+
+### Context Clicks
+- [ ] It should register a _link_ context menu item for opening sites in the sidebar
+- [ ] It should register a _page_ and _tab_ context menu item for opening sites in the sidebar
+- [ ] Both context menu buttons should include the extension icon
+
+### Sidebar Homepage
+- [ ] It should have a homepage
+- [ ] It should have disimissable onboarding content (text + icon for MVP)
```
Comment (ianb) [inline docs/acceptance.md]: #26. Is there a `?` link somewhere to bring back the onboarding?

File: docs/acceptance.md [hunk 20]
```diff
@@ -0,0 +1,42 @@
...
+- [ ] It should show a list of currently open tabs
+- [ ] It should show a list of recently sidebarred tabs
+- [ ] It should have a feedback button that links to an external survey with the query params `ver={add-on-version}`, `ref={sidebar}` and `rel={firefox-release}`
+- [ ] It should be titled with the extension name
```
Comment (ianb) [inline docs/acceptance.md]: #5 (fixed)

File: docs/acceptance.md [hunk 21]
```diff
@@ -0,0 +1,42 @@
...
+### Stretch
+- [ ] The sidebar tab should be openable by a browserAction. Currently blocked by[this bug](https://bugzilla.mozilla.org/show_bug.cgi?id=1427431).
+- [ ] The browser action should accept a fill value of `context-fill`
+
+### A11y
+- [ ] All buttons and links should have visible focus states
+- [ ] All buttons and links should be accessible via keyed entry (tab selection)
+- [ ] All form elements should include appropriate label attributes
+- [ ] All grouped buttons should be nested in a <fieldset> and described with a legend
+- [ ] All UI should be verified to use A11y friendly contrast ratios
```
Comment (ianb) [inline docs/acceptance.md]: I've put this a11y checklist in #27

**Comment by ianb** [comment]
Probably there should be an item about analytics

**Comment by ianb** [comment]
I put the references to existing tickets (from my comments) in 88706d48585222c0d28b34c418de48c6db152832

File: docs/acceptance.md [hunk 22]
```diff
@@ -0,0 +1,45 @@
...
+- [ ] It should inlude a refresh button to refresh the page. See #11
+- [ ] It should render websites using a common mobile header. See #7 and #24
+- [x] It should let users switch to a normal desktop header. See #8
+- [x] It should remember per site if users make the choice to switch headers. See #8
+
+### Global
+- [ ] It should extend the max-width of the sidebar to up to 50% of the total browser window. See #10
+
+### Stretch
+- [ ] The sidebar tab should be openable by a browserAction. Currently blocked by[this bug](https://bugzilla.mozilla.org/show_bug.cgi?id=1427431). See #23
```
Comment (pdehaan) [inline docs/acceptance.md]: Nit: add a space between "blocked by" and "[this bug]".

File: docs/acceptance.md [hunk 23]
```diff
@@ -0,0 +1,45 @@
...
+- [ ] It should match final visual specs.
+
+### Context Clicks
+- [x] It should register a _link_ context menu item for opening sites in the sidebar. See #3
+- [x] It should register a _page_ and _tab_ context menu item for opening sites in the sidebar. See #4
+- [ ] Both context menu buttons should include the extension icon
+
+### Sidebar Homepage
+- [ ] It should have a homepage
+- [ ] It should have disimissable onboarding content (text + icon for MVP). See #26
```
Comment (pdehaan) [inline docs/acceptance.md]: typo: disimissable

File: docs/acceptance.md [hunk 24]
```diff
@@ -0,0 +1,45 @@
+<img width="840px" height="auto" src="./acceptance-wires.png" alt="Splitting basics">
+
+### In General
+- [ ] It should generally match Photon styles.
+- [ ] It should have a way for users to provide feedback without leaving the experiment. See #6
+- [ ] It should match final visual specs.
```
Comment (pdehaan) [inline docs/acceptance.md]: Should this have a link to "#43; Apply visual designs"?

**Comment by pdehaan** [comment]
This is a great list!
I know I am ~~mildly~~ severely annoying, but should I file bugs for all the checklist items that _DON'T_ have bug numbers assigned to them, and then add a commit to this PR which links the bugs?
Stuff like "It should have a homepage" seems kind of big and serious, and possibly worth tracking in a bug so we have somewhere concrete to add comments/discussion/concerns.

**Comment by ianb** [comment]
Let's land this as it is, and update it as we go. @pdehaan if there's something that needs a bug, yeah. I put a bunch of bug references for the ones I know about.

**Pull request merged**